lib/kerror/index.js

```diff
@@ -62,11 +62,46 @@ function get (domain, subdomain, error, ...placeholders) {
     | domains[domain].subdomains[subdomain].code << 16
     | domains[domain].subdomains[subdomain].errors[error].code;
 
+  let kerror;
   if (kuzzleError.class === 'PartialError') {
-    return new errors[kuzzleError.class](message, body, id, code);
+    kerror = new errors[kuzzleError.class](message, body, id, code);
+  }
+  else  {
+    kerror = new errors[kuzzleError.class](message, id, code);
+  }
+
+  cleanStackTrace(kerror, 3);
+
+  return kerror;
+}
+
+/**
+ * Removes the n first lines of the stacktrace because they are related
+ * to internal mechanisms
+ *
+ * Eg:
+ *  at new PluginImplementationError (
+ *  at get (kuzzle/lib/kerror/index.js:70:14)
+ *  at Object.get (kuzzle/lib/kerror/index.js:146:38)
+ *  // Line that triggered the error =>
+ *  at ControllerManager.add (kuzzle/lib/core/framework/application.ts:226:34)
+ */
+function cleanStackTrace (error, depth) {
+  // Keep the original error message (2 lines for PluginImplementationError)
+  let errorMessage = '';
+  let i = error.name === 'PluginImplementationError' ? 2 : 1;
+  while (i--) {
+    const idx = error.stack.indexOf('\n') + 1;
+    errorMessage += error.stack.slice(0, idx);
+    error.stack = error.stack.slice(idx);
   }
 
-  return new errors[kuzzleError.class](message, id, code);
+  // Remove the first 3 lines of the stacktrace
```

Comment:
Instead of removing them, can we replace them with a single line telling that some lines have been removed?
Something like `[...Kuzzle internal calls deleted...]`

This is not entirely for aesthetics: I'm wondering if removing lines from the stacktrace is entirely safe, and if there won't be cases were these lines shouldn't be removed. In these cases, having some kind of information telling that the stacktrace is incomplete can help immensely when debugging.

Also, I think that `InternalError` errors shouldn't be cleaned up.

Reply:
I remove the lines only when we use our error wrapper, and every time we use it we have those 3 additional lines because the error object is instantiated here and not on the `throw` line.